controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,253 @@
+		Clients:             []ConnectedClient{},
+		StorageCluster: InfoStorageCluster{
+			NamespacedName:          types.NamespacedName{Name: storageCluster.Name, Namespace: storageCluster.Namespace},
+			StorageProviderEndpoint: storageCluster.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return string(yamlData), nil
+
+}
+
+func getStorageSystemName(storageCluster *ocsv1.StorageCluster) (string, error) {
+	storageSystemRef := util.Find(storageCluster.OwnerReferences, func(ref *metav1.OwnerReference) bool {
+		return ref.Kind == "StorageSystem"
+	})
+	if storageSystemRef != nil {
+		return storageSystemRef.Name, nil
+	}
+
+	return "", fmt.Errorf(
+		"failed to find parent StorageSystem's name in StorageCluster %q"+
+			" ownerreferences, %v",
+		storageCluster.Name,
+		storageCluster.OwnerReferences)
+
+}
+
+func getIsDROptimized(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (bool, error) {
+	var cephCluster rookCephv1.CephCluster
+	err := r.Client.Get(
+		r.ctx,
+		types.NamespacedName{
+			Name:      generateNameForCephClusterFromString(storageCluster.Name),
+			Namespace: storageCluster.Namespace},
+		&cephCluster)
+	if err != nil {
+		return false, err
+	}
+	if cephCluster.Status.CephStorage == nil || cephCluster.Status.CephStorage.OSD.StoreType == nil {
+		return false, fmt.Errorf("cephcluster %v status does not have OSD store information, %v", cephCluster.Name, storageCluster.Name)
+	}
+	bluestorerdr, ok := cephCluster.Status.CephStorage.OSD.StoreType["bluestore-rdr"]
+	if !ok {
+		return false, nil
+	}
+	total := getOsdCount(storageCluster, r.serverVersion)
+	if bluestorerdr < total {
+		return false, nil
+	}
+	return true, nil
+}
+
+func getOdfVersion(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	var csvs operatorsv1alpha1.ClusterServiceVersionList
+	err := r.Client.List(r.ctx, &csvs, client.InNamespace(storageCluster.Namespace))
+	if err != nil {
+		return "", err
+	}
+
+	csv := util.Find(csvs.Items, func(csv *operatorsv1alpha1.ClusterServiceVersion) bool {
+		return strings.HasPrefix(csv.Name, OdfOperatorNamePrefix)
+	})
+	if csv != nil {
+		return csv.Spec.Version.String(), nil
+	}
+
+	return "", fmt.Errorf("failed to find csv with prefix %q", OdfOperatorNamePrefix)
+}
+
+func getCephFsid(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	rookCephMonSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RookCephMonSecretName,
+			Namespace: storageCluster.Namespace,
+		},
+	}
+
+	if err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(rookCephMonSecret), rookCephMonSecret); err != nil {
+		return "", err
+	}
+	if val, ok := rookCephMonSecret.Data[FsidKey]; ok {
+		return string(val), nil
+	}
+
+	return "", fmt.Errorf("failed to fetch ceph fsid from %q secret", RookCephMonSecretName)
```

Comment:
SImilar to my last comment:

```suggestion
	if val, ok := rookCephMonSecret.Data[FsidKey]; !ok {
		return "", fmt.Errorf("failed to fetch ceph fsid from %q secret", RookCephMonSecretName)
	}
	
	return string(val), nil
```